lib/graphql/define/defined_object_proxy.rb

```diff
@@ -10,6 +10,11 @@ def types
         GraphQL::Define::TypeDefiner.instance
       end
 
+      # TODO: do I actually want to add this?
```

Comment:
It's definitely useful -- it's the basis for things like `field_defn.required_feature_flag` in your example above (that is, arbitrary data attached to schema members).

Right now it's only supported via 

```ruby
GraphQL::BaseType.accepts_definitions(
  required_feature_flag: GraphQL::Define.assign_metadata_key(:required_feature_flag)
)
```

and then 

```ruby
MyType = GraphQL::ObjectType.define do 
  required_feature_flag :top_secret_access 
end 
```

But with this you could "just" do 

```ruby 
MyType = GraphQL::ObjectType.define do 
  metadata :required_feature_flag, :top_secret_access
end 
```

I added it because I wanted to mess with `metadata` without extending every schema in the test suite! Maybe it would be useful IRL to have a helper like that too.